app/src/androidTest/java/com/github/se/project/ui/endToEnd/TutorFindLessonEndToEnd.kt

```diff
@@ -0,0 +1,367 @@
+    composeTestRule.onNodeWithTag("sectionDropdownItem-SC").performClick()
+    composeTestRule.onNodeWithTag("academicLevelDropdown").performClick()
+    composeTestRule.onNodeWithTag("academicLevelDropdownItem-BA3").performClick()
+    composeTestRule.onNodeWithTag("confirmButton").performClick()
+    assertEquals(Role.TUTOR, mockProfileViewModel.currentProfile.value?.role)
+
+    // Create Tutor Profile Screen
+    composeTestRule.onNodeWithTag("welcomeText").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("checkbox_FRENCH").performClick()
+    composeTestRule.onNodeWithTag("checkbox_ENGLISH").performClick()
+    composeTestRule.onNodeWithTag("subjectButton").performClick()
+    composeTestRule.onNodeWithTag("dropdownANALYSIS").performClick()
+    composeTestRule.onNodeWithTag("dropdownPHYSICS").performClick()
+    composeTestRule.onNodeWithTag("subjectButton").performClick()
+    composeTestRule.onNodeWithTag("priceSlider").performGesture { swipeRight() }
+    composeTestRule.onNodeWithTag("confirmButton").performClick()
+
+    // Create Tutor Schedule Screen
+    composeTestRule
+        .onNodeWithTag("welcomeText")
+        .assertTextEquals("John, show us your availabilities")
+    composeTestRule.onNodeWithTag("Slot_0_0").performClick()
+    composeTestRule.onNodeWithTag("Slot_0_3").performClick()
+    composeTestRule.onNodeWithTag("Slot_0_2").performClick()
+    composeTestRule.onNodeWithTag("Slot_0_6").performClick()
+    composeTestRule.onNodeWithTag("FindStudentButton").performClick()
+
+    // Home Screen
+    composeTestRule.onNodeWithContentDescription("Profile Icon").performClick()
+    composeTestRule.onNodeWithTag("profileAcademicInfo").assertTextEquals("SC - BA3")
+    composeTestRule.onNodeWithTag("profileInfoRole").assertTextEquals("Tutor")
+    composeTestRule.onNodeWithTag("lessonsCount").assertTextEquals("0")
+    composeTestRule.onNodeWithTag("closeButton").performClick()
+    composeTestRule.onNodeWithContentDescription("Profile Icon").assertExists()
+    composeTestRule.onNodeWithTag("middlePlus").performClick()
+    composeTestRule.onNodeWithTag("screenTitle").assertExists()
+    composeTestRule.onNodeWithText("Physics Tutoring").assertIsDisplayed()
+    composeTestRule.onNodeWithText("physics").assertIsDisplayed()
+    composeTestRule.onNodeWithText("physics").performClick()
+    // work
+    composeTestRule.waitForIdle()
+    composeTestRule.onNodeWithTag("tutorLessonResponseScreen").assertExists()
+    composeTestRule.onNodeWithText("Ozymandias Halifax").assertExists()
+    composeTestRule.onNodeWithTag("confirmButton").performClick()
+    composeTestRule.onNodeWithTag("confirmDialogConfirmButton").performClick()
+    composeTestRule.onNodeWithContentDescription("Profile Icon").assertExists()
+    composeTestRule
+        .onNodeWithTag("section_Waiting for the Student Confirmation")
+        .assertIsDisplayed()
+    composeTestRule.onNodeWithTag("lessonCard_0").assertExists()
```

Comment:
This test is really well done with the use of waitForIdle() to be sure the screen has finis his composition before continuing the tests.

However, the last part (after the comment // Home Screen at line 224), could have been more documented so that we could easily understand what is the flow we want to test.